File: web/src/types/budget.ts
export interface Budget {
  id: number
  year: number
  month: number
  amount: number
}

export interface BudgetProgress {
  year: number
  month: number
  budget: number | null
  spent: number
  remaining: number | null
  percent: number | null
  percent_actual: number | null
  is_over_budget: boolean | null
}

File: api/schemas/budget.py
"""Budget schemas."""

from typing import Optional
from pydantic import BaseModel, Field


class BudgetResponse(BaseModel):
    id: int
    year: int
    month: int
    amount: float

    class Config:
        from_attributes = True


class BudgetSet(BaseModel):
    amount: float = Field(..., gt=0)


class BudgetProgress(BaseModel):
    year: int
    month: int
    budget: Optional[float] = None
    spent: float
    remaining: Optional[float] = None
    percent: Optional[float] = None
    percent_actual: Optional[float] = None
    is_over_budget: Optional[bool] = None
